### PR TITLE
chore: remove reference to `prefect.yaml`, which was removed in the previous PR

### PR DESCRIPTION
**Pull request by jamiezieziula**
Relates to PLA-309

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
         if: github.ref_type == 'tag'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: gh release upload ${{ github.ref_name }} prefect-crds.yaml prefect.yaml
+        run: gh release upload ${{ github.ref_name }} prefect-crds.yaml
```
Comment (mitchnielsen) [inline .github/workflows/release.yaml:55]: I missed this too - previously, `prefect-crds.yaml` had only the CRDs, and `prefect.yaml` had the CRDS _and_ the Deployment, etc, so folks could add it all in one. We could still populate this by running `helm template` and piping it out to a file, but I'm fine skipping this for now as I bet most will install it via helm.

**Review by mitchnielsen — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jamiezieziula) [reply .github/workflows/release.yaml:55]: i'm happy to do that if you think thats useful!

Comment (mitchnielsen) [reply .github/workflows/release.yaml:55]: can just be a follow-up issue i think, i don't see it as a big priority unless users request it

Comment (jamiezieziula) [reply .github/workflows/release.yaml:55]: follow up issue [here](https://linear.app/prefect/issue/PLA-334/update-prefect-operator-to-package-manifests-alongside-release)

**Pull request merged**